### PR TITLE
feat: Include triggering rules in webhook payload

### PR DESCRIPTION
**Pull request by ankurd1**
### Context
At Flexport, we are heavy users of Sentry webhooks. We use these to assign, prioritize and route alerts through various systems. We have a bunch of different "alert rules" in the sentry UI which all fire the same webhook. But in the webhook, we cannot distinguish which particular rule fired. We need this information because it's an important input for prioritization.

### Solution
The solution is pretty simple. I just added code to wire the triggering rule names to the webhook payload.

### Test plan
- Run sentry locally and verify that the payload includes this field
- Update tests

### CODE REVIEW

File: src/sentry/plugins/bases/notify.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def get_plugin_type(self):
 
     def notify(self, notification):
         event = notification.event
-        return self.notify_users(event.group, event)
+        return self.notify_users(event.group, event, map(lambda r: r.label, notification.rules))
```
Comment (markstory) [inline src/sentry/plugins/bases/notify.py]: Could use a list comprehension here

```suggestion
        return self.notify_users(event.group, event, [r.label for r in notification.rules])
```

**Review by dcramer — APPROVED** [review]

**Comment by dcramer** [comment]
Thanks!

**Comment by legalosLOTR** [comment]
Thanks for approving guys! When can we expect this to get deployed?

File: src/sentry/plugins/bases/notify.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def get_plugin_type(self):
 
     def notify(self, notification):
         event = notification.event
-        return self.notify_users(event.group, event)
+        return self.notify_users(event.group, event, [r.label for r in notification.rules])
```
Comment (dcramer) [inline src/sentry/plugins/bases/notify.py]: Could we change this to kwargs?

```suggestion
        return self.notify_users(event.group, event, triggering_rules=[r.label for r in notification.rules])
```

File: src/sentry/plugins/sentry_webhooks/plugin.py [hunk 3]
```diff
@@ -103,7 +104,7 @@ def send_webhook(self, url, payload):
             verify_ssl=False,
         )
 
-    def notify_users(self, group, event, fail_silently=False):
-        payload = self.get_group_data(group, event)
+    def notify_users(self, group, event, triggering_rules, fail_silently=False):
```
Comment (dcramer) [inline src/sentry/plugins/sentry_webhooks/plugin.py]: Just to future proof (and what I'm going to check in other plugins) could we add kwargs support?

```suggestion
    def notify_users(self, group, event, triggering_rules, fail_silently=False, **kwargs):
```

**Review by dcramer — CHANGES_REQUESTED** [review]
Would you mind making a few quick changes and I'll take some time to confirm this won't break any other existing integrations (or ill fix them ahead of landing this)?

**Comment by dcramer** [comment]
I think there's a low impact chance on this since:

1. Old Slack is fine
2. New Slack is fine
3. There's not many plugins which would even use the notify_users hook

So as long as we do the kwargs changes I think we're safe to merge, since I commonly used the kwargs pattern in a lot of endpoints to future proof them.

**Comment by legalosLOTR** [comment]
Made the requested changes. I also grepped for `notify_users` and these are the only files that mention it.

**Review by dcramer — APPROVED** [review]

**Comment by dcramer** [comment]
Thanks!

**Pull request merged**